import printer.(println);

instance Exception (Int);

record Test (
    x: Int32
);

overload Test() = Test(-1);

overload Test(x:Int32) --> returned:Test {
    println("Created ", x);
    returned.x = x;
}

overload Test(other:Test) --> returned:Test {
    println("Copied ", other.x);
    returned.x = other.x;
}

overload destroy(t:Test) {
    println("Destroyed ", t.x);
}

main () {
    try {
        var a = Test(1);
        if (true)
            throw 1;
        var b = Test(2);
    }
    catch (e) {
        println("Caught!");
    }
}
